src/createServer.js

```diff
@@ -1,11 +1,211 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+  let users = [];
+  let expenses = [];
+
+  app.use(express.json());
+
+  app.get('/users', (req, res) => {
+    res.statusCode = 200;
+
+    res.send(users);
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    const findUser = users.find((user) => user.id === +id);
+
+    if (!findUser) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.statusCode = 200;
+    res.send(findUser);
+  });
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const newUser = {
+      id: users.length + 1,
+      name,
+    };
+
+    users.push(newUser);
+    res.statusCode = 201;
+    res.send(newUser);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    const filtereedUser = users.filter((user) => user.id !== +id);
+
+    if (filtereedUser.length === users.length) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    users = filtereedUser;
+    res.sendStatus(204);
+  });
+
+  app.patch('/users/:id', (req, res) => {
+    const { id } = req.params;
+
+    const findUser = users.find((user) => user.id === +id);
+
+    if (!findUser) {
+      res.sendStatus(404);
+    }
+
+    const { name } = req.body;
+
+    Object.assign(findUser, { name });
+
+    res.send(findUser);
+    res.statusCode = 200;
+  });
+
+  app.post('/expenses', (req, res) => {
+    const {
+      userId,
+      spentAt,
+      title,
+      amount,
+      category,
+      note,
+    } = req.body;
```

Comment:
You could destructure only userId here
```suggestion
    const { userId } = req.body;
```
As I understand, you need only it. And in `newExpenses` use `...req.body`.